client/src/api/v2/camera.js

```diff
@@ -39,3 +41,140 @@ export function useOverlays(device) {
 
   return { overlays, setActiveOverlay };
 }
+
+/**
+ * Send a message to server to start recording video
+ */
+export function startRecording() {
+  emit('start-camera-recording');
+}
+
+/**
+ * Send a message to server to stop recording video
+ */
+export function stopRecording() {
+  emit('stop-camera-recording');
+}
+
+/**
+ * Get the last MQTT payload for the status of a camera recording
+ *
+ * @param {string} device Camera
+ * @returns {string} Payload
+ */
+export function getLastPayload(device) {
+  return sessionStorage.getItem(`camera-recording-last-payload-${device}`);
+}
+
+/**
+ * Store the last MQTT payload for the status of a camera recording
+ *
+ * @param {string} device Camera
+ * @param {string} payloadString String representation of payload
+ * @returns {string} Payload
+ */
+export function storeLastPayload(device, payloadString) {
+  return sessionStorage.setItem(`camera-recording-last-payload-${device}`, payloadString);
+}
+
+/**
+ * Parse the payload to get the status
+ *
+ * @param {string} payload Payload
+ * @returns {string} Status
+ */
+export function getStatusFromPayload(payload) {
+  const data = JSON.parse(payload);
+  if (!data) return null;
+  return data.status;
+}
+
+/**
+ * Parse the payload to get the information (fields other than status)
+ *
+ * Payload structure is defined in the 'V3 MQTT Topics' page on Notion.
+ * Topic is /v3/camera/recording/status/<primary/>secondary>.
+ *
+ * @param {string} payload Payload
+ * @returns {React.Component} Component
+ */
+export function getInfoFromPayload(payload) {
+  const data = JSON.parse(payload);
+  if (!data) return null;
+
+  const format = (field) => {
+    let name = startCase(field);
+    let value = '';
+
+    // format field value
+    switch (field) {
+      case 'diskSpaceRemaining':
+        value = formatBytes(data.diskSpaceRemaining);
+        break;
+
+      case 'recordingMinutes':
+        {
+          const mins = Math.floor(data.recordingMinutes);
+          const secs = (data.recordingMinutes - mins) * 60;
+          value = `${mins} minutes ${secs} seconds`;
+          name = 'Recording Time';
+        }
+        break;
+
+      default:
+        value = data[field];
+        break;
+    }
+
+    return `${name}: ${value}`;
+  };
+
+  return (
```

Comment:
Methods in the API just transform data. The rendering part of this function should instead be a separate display component.

Reply:
Done
8a8d1bc